ExampleApp/Assets/OptimoveSdk/Editor/SetUpXcodeProject.cs

```diff
@@ -146,16 +152,135 @@ private static void AddUnityVersionToPlist(string buildPath)
+    {
+        string src = Path.Combine(Application.dataPath, "OptimoveNativeAssets~", frameworkName);
+        string dest = Path.Combine(pathToBuiltProject, frameworkName);
+
+        if (Directory.Exists(dest)){
+            return;
+        }
+
+        if (!Directory.Exists(src)){
+            throw new BuildFailedException("Please, add OptimoveNativeAssets~ to Assets folder");
+        }
+
+        CopyDirectory(src, dest);
+
+        string fileGuid = project.AddFile(dest, Path.Combine("Frameworks", frameworkName));
+
+        //MAIN TARGET
+        //embed and sign
+        project.AddFileToEmbedFrameworks(mainTargetGuid, fileGuid);
+
+        //link binary
+        string mPhaseGuid = project.GetFrameworksBuildPhaseByTarget(mainTargetGuid);
+        project.AddFileToBuildSection(mainTargetGuid, mPhaseGuid, fileGuid);
+
+        //UNITY FRAMEWORK TARGET
+
+        //link binary
+        string uPhaseGuid = project.GetFrameworksBuildPhaseByTarget(secondaryTargetGuid);
+        project.AddFileToBuildSection(secondaryTargetGuid, uPhaseGuid, fileGuid);
+    }
+
+    private static void CopyDirectory(string sourcePath, string destPath)
+    {
+        Directory.CreateDirectory(destPath);
+
+        foreach (string file in Directory.GetFiles(sourcePath))
+            File.Copy(file, Path.Combine(destPath, Path.GetFileName(file)));
+
+        foreach (string dir in Directory.GetDirectories(sourcePath))
+            CopyDirectory(dir, Path.Combine(destPath, Path.GetFileName(dir)));
+    }
+
+    private static void MaybeSetUpOptimoveNotificationServiceExtension(PBXProject project, string mainTargetGuid, string pathToBuiltProject, string projectPath)
+    {
+        string optimobileCredentials = GetValueFromOptimovePlist("optimoveMobileCredentials");
+        if (optimobileCredentials == null){
+            return;
+        }
+
+        string extensionTargetName = "OptimoveUnityNotificationServiceExtension";
+        string extensionGuid = project.TargetGuidByName(extensionTargetName);
+        if (!string.IsNullOrEmpty(extensionGuid)){
+            return;
+        }
+
+        string extensionFilesPath = Path.Combine(Application.dataPath, "OptimoveNativeAssets~", "NotificationServiceExtension");
+
+        // add plist
+        string extensionPath = Path.Combine(pathToBuiltProject, extensionTargetName);
+        Directory.CreateDirectory(extensionPath);
+        string plistPath = Path.Combine(extensionPath, "Info.plist");
+        PlistDocument notificationServicePlist = new PlistDocument();
+        notificationServicePlist.ReadFromFile(Path.Combine(extensionFilesPath, "Info.plist"));
+        notificationServicePlist.WriteToFile(plistPath);
+
+        // add target
+        extensionGuid = project.AddAppExtension(mainTargetGuid,
+            extensionTargetName,
+            PlayerSettings.GetApplicationIdentifier(BuildTargetGroup.iOS) + "." + extensionTargetName,
+            extensionTargetName + "/Info.plist"
```

Comment:
Path combine?

Reply:
this is a setting used by xcode, so, theoretically always requires '/'